snippets/base/templates/admin/login.html

```diff
@@ -0,0 +1,62 @@
+{% extends "admin/base_site.html" %}
+{% load i18n admin_static %}
+
+{% block extrastyle %}{{ block.super }}<link rel="stylesheet" type="text/css" href="{% static "admin/css/login.css" %}" />{% endblock %}
+
+{% block bodyclass %}{{ block.super }} login{% endblock %}
+
+{% block usertools %}{% endblock %}
+
+{% block nav-global %}{% endblock %}
+
+{% block content_title %}{% endblock %}
+
+{% block breadcrumbs %}{% endblock %}
+
+{% block content %}
+  {% if form.errors and not form.non_field_errors %}
+    <p class="errornote">
+      {% if form.errors.items|length == 1 %}{% trans "Please correct the error below." %}{% else %}{% trans "Please correct the errors below." %}{% endif %}
+    </p>
+  {% endif %}
+
+  {% if form.non_field_errors %}
+    {% for error in form.non_field_errors %}
+      <p class="errornote">
+        {{ error }}
+      </p>
+    {% endfor %}
+  {% endif %}
+
+  <div id="content-main">
+    {% if settings.OIDC_ENABLE %}
+      <p style="text-align:center; font-weight: bold; font-size: 150%">
+        <a href="{% url 'oidc_authentication_init' %}">Mozilla SSO</a>
+      </p>
+    {% else %}
+      <form action="{{ app_path }}" method="post" id="login-form">{% csrf_token %}
+        <div class="form-row">
+          {{ form.username.errors }}
+          {{ form.username.label_tag }} {{ form.username }}
+        </div>
+        <div class="form-row">
+          {{ form.password.errors }}
+          {{ form.password.label_tag }} {{ form.password }}
+          <input type="hidden" name="next" value="{{ next }}" />
+        </div>
+        {% url 'admin_password_reset' as password_reset_url %}
+        {% if password_reset_url %}
+          <div class="password-reset-link">
+            <a href="{{ password_reset_url }}">{% trans 'Forgotten your password or username?' %}</a>
+          </div>
+        {% endif %}
+        <div class="submit-row">
+          <label>&nbsp;</label><input type="submit" value="{% trans 'Log in' %}" />
+        </div>
+      </form>
+    {% endif %}
+    <script type="text/javascript">
+     document.getElementById('id_username').focus()
```

Comment:
I'm sure this is just standard Django login form, but instead of inline JS we could just use the [autofocus HTML5 attribute](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#attr-autofocus).

Not a blocker, just a suggestion.

Reply:
Indeed that's the standard django admin login page. I'd like to implement autofocus but I guess we need to either manual write the html for username input field or alter the widget on the django level. 

Is there a simpler build-in way to archive this?

Reply:
BTW I'm going to merge this today as is to get oidc live but I don't mean to stop this discussion.

Reply:
Cool. I thought I remembered a template filter that would add an attribute to a django field, but now I can't find it. This is certainly not a big deal, but it would be nice :) Especially if you want to turn on CSP at some point.

Reply:
Good point on CSP, I took a TODO to fix this. Thanks!